acpi.adoc

```diff
@@ -1,26 +1,142 @@
+* Be 64-bits clean.
+** RSDT must not be implemented.
+** 32-bit address fields must be 0.
+* Implement a firmware-first APEI model (cite:[ACPI] Section 18.4)
+
+=== ACPI-specific Hardware Requirements
+
+This section lists the BRS-specific hardware requirements imposed by
+ACPI support.
+
+==== HW-Reduced ACPI Model
+
+Compliant RISC-V platforms only implement the HW-Reduced ACPI Model. This means the hardware portion of
+cite:[ACPI] (Section 4) is not required or supported. All functionality is instead provided through equivalent
+software-defined interfaces and the complexity in supporting ACPI is reduced.
+
+The following additional BRS requirements apply:
+
+* Both interrupt-signaled and GPIO-signaled ACPI Events are supported
+  (cite:[ACPI] Section 5.6)
+* There is at least 1 wake event.
+** Note: for systems that do not support Sx states except S5 soft off, this can be just the power button.
+* For the ACPI Platform Error Interfaces (APEI) support:
+** There is 1 event for non-fatal error signaling (cite:[ACPI] Section 18.3.2.7.2)
+** There is 1 NMI-equivalent signal for use in fatal errors (cite:[ACPI] Section 18)
+
+==== Hart Performance Control
+
+If the platform provides support for OS-directed hart performance control, then it must be exposed using
+Collaborative Processor Performance Control (CPPC, cite:[ACPI] Section 8).
+
+==== Time and Alarm Device
+
+UEFI Runtime Service should be used to provide time services to the
+OS, unless the RTC is subject to arbitration issues between concurrent
+OS and UEFI accesses to the underlying hardware. See <<<UEFI Real-Time
+Clock>>>.
+
+In situations where the TAD device depends on a vendor-specific OS
+driver for correct function (SPI, I2C, etc), the TAD must be
+function if the OS driver is not loaded. E.g. when a dependent
+driver is loaded, an AML method switches further accesses to go
+through the driver-backed OperationRegion.
+
+=== Additional ACPI Table Requirements
+
+This section lists additional BRS-specific requirements
+for the mandatory and conditionally-required ACPI tables in a compliant
+platform.
+
+This section does not list ACPI tables without additional BRS-imposed
+requirements.
+
+All ACPI tables not listed here may be used, if they comply with the
+syntax and semantics of the ACPI or other relevant firmware specification.
+
+==== MADT
+
+Entry ordering can be correlate with initialization order by an OS, but
+should not be taken to reflect affinity in resource sharing,
+e.g. sockets, caches, etc. RINTC hart ID and ACPI Processor UID should
+not be decoded in a platform-specific manner to divine CPU topology.
+The PPTT Processor Properties Topology Table (PPTT) is to be used to
+describe affinities.
+
+==== PPTT
+
+The Processor Properties Table (PPTT, cite:[ACPI] Section 5.2.29)
+describes the hart affinities and shared resources, such as caches.
+It is mandatory, even for platforms with a simple hart topology.
+
+==== Conditionally Required ACPI Tables
+
```

Comment:
What about NUMA-related tables like SRAT/SLIT.

Reply:
Yes, thanks again for helping me to understand the philosophy behind adding things in BRS spec vs Industry standard.